File: JavaScript/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
    const words = s.split(' ').filter(item => item.length > 0);
    const n = words.length;
    let ans = '';
    for (let i = n - 1; i > 0; --i) {
        ans += words[i];
        ans += ' ';
    }
    if (n > 0) {
        ans += words[0];
    }
    return ans;
};

File: C++/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stack<string> words;
        int n = s.length();
        int i = 0;
        while (i < n) {
            if (s[i] != ' ') {
                // string word = s[i];
                string word = "";
                while (i < n && s[i] != ' ') {
                    word += s[i];
                    ++i;
                }
                words.push(word);
            }
            ++i;
        }
        string ans = "";
        while (words.size() > 1) {
            ans += words.top();
            ans += ' ';
            words.pop();
        }
        if (!words.empty()) {
            ans += words.top();
        }
        return ans;
    }
};